src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -98,7 +98,7 @@ CancellationToken cancellationToken
                         if (!string.IsNullOrEmpty(csdlFilter))
                         {
                             XslCompiledTransform transform = GetFilterTransform();
-                            stream = ApplyFilter(csdl, csdlFilter, transform);
+                            stream = csdl.StartsWith("http") ? ApplyFilter(stream, csdlFilter, transform) : ApplyFilter(csdl, csdlFilter, transform);
```

Comment:
```suggestion
                            stream = csdl.StartsWith("http", StringComparison.OrdinalIgnoreCase) ? ApplyFilter(new StreamReader(stream), csdlFilter, transform) : ApplyFilter(new StreamReader(csdl), csdlFilter, transform);
```

Reply:
The original code here seems wrong.  GetStream does all the logic to determine if stream  is coming from a file or via HTTP.  Currently the code is getting the stream and then ignoring it if there is a filter.  The solution should be to change ApplyFilter to just accept the stream that comes from GetStream.